### PR TITLE
adding pypi automatic deployment

### PR DESCRIPTION
**Pull request by azaf**
It just needs two variables to be defined in travis configuration named "test_pypi_USER" and "test_pypi_PASS" in order to work correctly.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -47,8 +47,14 @@ jobs:
...
-        script: /bin/bash scripts/docker_push.sh "${TRAVIS_TAG}" latest
+        - provider: script
+          on:
+            tags: true
+          script: /bin/bash scripts/docker_push.sh "${TRAVIS_TAG}" latest
+        - provider: pypi
+          user: ${test_pypi_USER}
+          password: ${test_pypi_PASS}
+          on:
+            branch: master
```
Comment (aperrin66) [inline .travis.yml]: We will want to build the package when a tag (=release) is created, not for every push to master :)

**Comment by akorosov** [comment]
@azaf , thank you for the PR! However more work seem to be required.
First, the building of Nansat by pypi seems to be not tested in your repository. 
If you look at the log:
https://travis-ci.org/azaf/nansat/builds/656298023?utm_medium=notification&utm_source=github_status

You will notice that it only ran one stage - Unit tests.
Please try to add a tag to your Nansat repository and test if Travis can build the Nansat package and attempt to submit it to Pypi.

It will definitely give an error at a later stage because it cannot push Nansat to Pypi withou correct credentials. But that is OK - we will fix it after the PR is merged.

**Comment by akorosov** [comment]
@azaf , please also see my consideraions here:
https://github.com/nansencenter/nansat/issues/458

**Comment by akorosov** [comment]
OK, I saw in this log:
https://travis-ci.org/azaf/nansat/jobs/656278294?utm_medium=notification&utm_source=github_status

that the package was actually created and no compilation is needed. This is fine then!

But I have one more request. Can you make a separate stage for Pypi deployment? So fa it is mixed in one stage which is called Build docker image and is confusing.

**Comment by aperrin66** [comment]
> OK, I saw in this log:
> https://travis-ci.org/azaf/nansat/jobs/656278294?utm_medium=notification&utm_source=github_status
> 
> that the package was actually created and no compilation is needed. This is fine then!
> 
> But I have one more request. Can you make a separate stage for Pypi deployment? So fa it is mixed in one stage which is called Build docker image and is confusing.

We could also rename the stage as "Build" and have two jobs in it, one for the Docker image, and one for the PyPI package. That way the two jobs would be run in parallel. (Stages are run sequentially, but jobs within stages are run in parallel)

File: .travis.yml [hunk 2]
```diff
@@ -47,8 +47,14 @@ jobs:
...
-        on:
-          tags: true
-        script: /bin/bash scripts/docker_push.sh "${TRAVIS_TAG}" latest
+        - provider: script
+          on:
+            tags: true
+          script: /bin/bash scripts/docker_push.sh "${TRAVIS_TAG}" latest
+        - provider: pypi
+          user: ${test_pypi_USER}
+          password: ${test_pypi_PASS}
```
[the remark this answers is not in the document]
Comment (akorosov) [reply .travis.yml]: It would also be nice to add PYPI_TOKEN to the short README section in the header

**Comment by azaf** [comment]
OK. The jobs are separated! Now I work on dealing with token instead of username and passwork!

**Comment by azaf** [comment]
OK. Now the result of travis says "Invalid or non-existent authentication information.". I think it is done and now ready for authentication with correct credential.

I have added an "If" line in comment for making this deployment similar to docker deployment. Please, uncommet that line in order to work for the same situations. It is now deploying without that "if" statement.

**Comment by azaf** [comment]
Let me update the readme! I almost forgot it!

**Comment by azaf** [comment]
Please check the last commit a8f01f3

File: .travis.yml [hunk 3]
```diff
@@ -7,14 +7,15 @@ services:
 # - DOCKER_ORG: the docker hub organization (or user) to which the image will be pushed
 # - DOCKER_USER: the docker hub user used to log in to the docker hub
 # - DOCKER_PASS: the password of this user
+# - PYPI_TOKEN   the token assosiated to pypi account for deployment. More information can be found here: https://pypi.org/help/#apitoken
```
Comment (aperrin66) [inline .travis.yml]: Small typo :)

Comment (azaf) [reply .travis.yml]: I am not a native ENG speaker. Thanks for notice!  :)

**Comment by azaf** [comment]
Please check the last commit cfd4b86

File: .travis.yml [hunk 4]
```diff
@@ -47,8 +48,19 @@ jobs:
...
-          tags: true
-        script: /bin/bash scripts/docker_push.sh "${TRAVIS_TAG}" latest
-...
\ No newline at end of file
+        - provider: script
+          on:
+            tags: true
+          script: /bin/bash scripts/docker_push.sh "${TRAVIS_TAG}" latest
+    - stage: 'pypi deployment'
+      #if: type = pull_request OR tag IS present  ###uncomment this line in order to work for specific cases similar to ""stage: 'Build docker image'""
```
Comment (aperrin66) [inline .travis.yml]: We don't need the same mechanism as for the Docker build. The reason is that for Docker, we build the image on pull requests and on tags. The image is then pushed only on tags. This enables us to check for problems in the image building process when a pull request is made.

For PyPI, the deployment provider both builds the package and uploads it to PyPI, so we are going to do this only on tags. So the only condition which is necessary is the `on: tags: true` statement in the deployment provider configuration.

Comment (azaf) [reply .travis.yml]: I have added what you want.  :)

Comment (azaf) [reply .travis.yml]: Alright. I have modified it.

Comment (azaf) [reply .travis.yml]: Please have a look @aperrin66

Comment (aperrin66) [reply .travis.yml]: It looks all right to me, the `if: tag IS present` statement might be redundant but it doesn't hurt :)
Can you just remove the empty line at the end of the file?

Comment (azaf) [reply .travis.yml]: It is done. I removed them. and push it  on Commit 9660ec9

**Comment by aperrin66** [comment]
It is fine now, I will just squash the commits so the git history is nice and pretty :)

**Pull request merged**